pennylane_lightning/src/Gates.cpp

```diff
@@ -89,7 +98,8 @@ const vector<CplxType> Pennylane::XGate::matrix{
     0, 1,
     1, 0 };
 
-void Pennylane::XGate::applyKernel(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices) {
+void Pennylane::XGate::applyKernel(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices, bool inverse) {
+    (void)inverse;  // gate is its own inverse
```

Comment:
Hmm, how come we didn't have such warning for `applyKernel`? 🤔 We had a thread with Thomas on this here: https://github.com/PennyLaneAI/pennylane-lightning/pull/85#discussion_r592480332.

Reply:
Yes, I think it's motivated for two reasons:
- So we won't get the warnings: `pennylane_lightning/src/Gates.cpp:101:147: warning: unused parameter ‘inverse’ [-Wunused-parameter]`
- I like having something that explicitly says the gate is it's own inverse. In other words, we're saying that we know about the `inverse` argument and don't need to do anything in this case, rather than we just forgot about it.

It'll work either way, so if we're opposed to having this line then I can remove. I don't think this has any performance implications, seems like something more for the compiler.

Reply:
Got it! It's odd that this warning is emitted 🤔 This is a personal opinion, but the statement looks confusing to me, as it's type casting a boolean to `void` without any effect. To me, having the parameter unnamed would make up for the unused parameter. The comment is indeed nice to indicate that the gate is its own inverse.

Reply:
Ok, updated to just have a commend and an unnamed param: fcd1c29a6755c013b916a651e2b724c834f0db9c